erc20-bridge-token/contracts/BridgeToken.sol

```diff
@@ -0,0 +1,68 @@
+pragma solidity ^0.8;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+
+contract BridgeToken is ERC20 {
+
+    address private _controller; // has CONTROL role
+
+    string private _name;
+    string private _symbol;
+    uint8 private _decimals;
+
+    uint64 _metadataLastUpdated;
+
+    constructor(
+        string memory name_, 
+        string memory symbol_,
+        uint8 decimals_
+    ) ERC20(name_, symbol_ ) {
+        _controller = msg.sender;
+
+        _name = name_;
+        _symbol = symbol_;
+        _decimals = decimals_;
+    }
+
+    function set_metadata(
+        string memory name_, 
+        string memory symbol_,
+        uint8 decimals_,
+        uint64 blockHeight_
+    ) public {
+        require(msg.sender == _controller, "ERR_NOT_CONTROLLER");
+        require(blockHeight_ >= _metadataLastUpdated, "ERR_OLD_METADATA");
+
+        _metadataLastUpdated = blockHeight_;
+        _name = name_;
+        _symbol = symbol_;
+        _decimals = decimals_;
+    }
+
+    function mint(address beneficiary, uint256 amount) public {
+        require(msg.sender == _controller, "ERR_NOT_CONTROLLER");
+        _mint(beneficiary, amount);
+    }
+
+    function burn(address act, uint256 amount) public {
+        require(msg.sender == _controller, "ERR_NOT_CONTROLLER");
+        _burn(act, amount);
+    }
```

Comment:
These methods should be pausable

Reply:
Pausable now. Latest token version: https://github.com/ilblackdragon/rainbow-token-connector/commit/2a9b5ceceb14ac7cccb126cf2d3cf5011e9dfdb1